### DISCUSSION TITLE
Generating an empty XML node with attribute.

### DISCUSSION THREAD

**Discussion by nikhil-warvatkar** [spatie/array-to-xml#203] [Q&A] [answered]
Hello everybody,
I need to generate an xml node with empty value, but it must have some attributes.
For example:
`<Header messageId= '123abc-cba' isTest='false' senderId='SND12345' receiverId='REC12345' security="security-key" originalMessageId='' errorCode='' errorMessage=''/>`

Result need for example:

> <order xmlns="https://github.com/spatie/array-to-xml">
	<Header messageId= '123abc-cba' isTest='false' senderId='SND12345' receiverId='REC12345' security="security-key" originalMessageId='' errorCode='' errorMessage=''/>
	<OrderType>DECORATED</OrderType>
	<PurchaseOrderNumber>TESTDECORDER-01</PurchaseOrderNumber>
	<Currency>EUR</Currency>
	<custom-key:1 attr1="value1" attr2="value2">
		<name>Vladimir</name>
		<nickname>greeflas</nickname>
	</custom-key:1>
	<custom-key:2>
		<name>Marina</name>
		<nickname>estacet</nickname>
	</custom-key:2>
</order>

It is possible to achieve this result with this library,
thank you all in advance

**Comment by weirdan** [comment, marked as answer]
Just add attributes as you would do normally:

```php
(new Spatie\ArrayToXml\ArrayToXml([
     'Header' => [
        '_attributes' => [
            'messageId' => '123abc-cba',
            'isTest' => 'false',
            //.......
        ],
     ],
]))->prettify()->toXml();
```

**Reply by nikhil-warvatkar** [reply]
Thanks for the early response.
this not worked with below code. Its wrapping around the below array.
$array = [
    'header' => [
        '_attributes' => ['messageId' => '123abc-cba', 'senderId' => '[CUSTOMERID]', 'receiverId' => 'PF', 'security' => 'securityID'],
    ],
    'custom-key:1' => [
        '_attributes' => ['attr1' => 'value1', 'attr2' => 'value2'],
        'name' => 'Vladimir',
        'nickname' => 'greeflas',
    ]
];

**Reply by weirdan** [reply]
Looks fine to me:
```php
require "vendor/autoload.php";

$array = [
    'header' => [
        '_attributes' => ['messageId' => '123abc-cba', 'senderId' => '[CUSTOMERID]', 'receiverId' => 'PF', 'security' => 'securityID'],
    ],
    'custom-key:1' => [
        '_attributes' => ['attr1' => 'value1', 'attr2' => 'value2'],
        'name' => 'Vladimir',
        'nickname' => 'greeflas',
    ]
];

echo (new Spatie\ArrayToXml\ArrayToXml($array))->prettify()->toXml();
```
and here's the result I got:
```xml
<?xml version="1.0"?>
<root>
  <header messageId="123abc-cba" senderId="[CUSTOMERID]" receiverId="PF" security="securityID"/>
  <custom-key:1 attr1="value1" attr2="value2">
    <name>Vladimir</name>
    <nickname>greeflas</nickname>
  </custom-key:1>
</root>
```

**Reply by nikhil-warvatkar** [reply]
ok. Thank you @weirdan

**Comment by nikhil-warvatkar** [comment]
Hello everybody,
How can I create a node with attribute and element value ?
E.g. 
`<OrderNode isOrderComplete="false">252.5</OrderNode>`

**Reply by weirdan** [reply]
It's in the docs: https://github.com/spatie/array-to-xml#adding-attributes

```php
$array = [
  'OrderNode' => [
      '_attributes' => ['isOrderComplete' => 'false'],
      '_value' => '252.5', // <===== note that it has to be a string. '252.5' works, 252.5 doesn't
   ],
];
```
